apps/src/applab/designElements/PropertyRow.jsx

```diff
@@ -61,7 +94,8 @@ var PropertyRow = React.createClass({
         type={this.props.isNumber ? 'number' : undefined}
         value={this.state.value}
         onChange={this.handleChangeInternal}
-        style={rowStyle.input} />;
+        onBlur={this.props.isIdRow ? this.onIdRowBlur : null}
```

Comment:
The argument I would make in favor of the other approach (and I don't feel super strongly about it) is that limiting the amount of logic that happens in the JSX makes it easier to quickly scan the component and understand what the output (to the DOM) is.